src/sentry/services/hybrid_cloud/user/__init__.py

```diff
@@ -113,9 +114,22 @@ class UserFilterArgs(TypedDict, total=False):
     emails: List[str]
 
 
-class UserService(
-    FilterQueryInterface[UserFilterArgs, RpcUser, UserSerializeType], InterfaceWithLifecycle
-):
+class UserService(InterfaceWithLifecycle):
+    @abstractmethod
+    def serialize_many(
+        self,
+        *,
+        filter: UserFilterArgs,
+        as_user: Optional[RpcUser] = None,
+        auth_context: Optional["AuthenticationContext"] = None,
```

Comment:
This string type hint might be a problem for Future Ryan (actually, [Parallel Branch](https://github.com/getsentry/sentry/pull/45447) Ryan) but it should be an easy fix. Worst case scenario, we push `AuthenticationContext` from `hybrid_cloud/auth/__init__.py` down to `hybrid_cloud/__init__.py`.

Reply:
Also, and just to be clear, *there is still a way to ask pydantic to resolve forward refs anyways*.  we may still have to consider adding that call in the right place if we run out of options.